framework/src/base/NonlinearSystemBase.C

```diff
@@ -1847,7 +1849,10 @@ NonlinearSystemBase::computeJacobianInternal(SparseMatrix<Number> &  jacobian)
     } // end loop over boundary nodes
 
     // Set the cached NodalBC values in the Jacobian matrix
```

Comment:
This comment belonged to the line `_fe_problem.assembly(0).setCachedJacobianContributions(jacobian);` which is now in the else branch.  If `kernel_type == Moose::KT_EIGEN` this comment would be a lie. Why are we dropping the cached values when `kernel_type == Moose::KT_EIGEN`? Good opportunity for an explanation in a comment...